client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,278 @@
+        ));
+    };
+    // get current blockhash
+    let original_block_hash = rpc_client.get_latest_blockhash().await?;
+    let block_hash_rw = Arc::new(RwLock::new(original_block_hash));
+    let current_block_height = Arc::new(AtomicU64::new(0));
+
+    let progress_bar = spinner::new_progress_bar();
+    progress_bar.set_message("Setting up...");
+
+    let exit_signal = Arc::new(AtomicBool::new(false));
+
+    // blockhash anc blockheight update task
+    {
+        let rpc_client = rpc_client.clone();
+        let block_hash_rw = block_hash_rw.clone();
+        let exit_signal = exit_signal.clone();
+        let current_block_height = current_block_height.clone();
+        tokio::spawn(async move {
+            while !exit_signal.load(std::sync::atomic::Ordering::Relaxed) {
+                if let Ok(blockhash) = rpc_client.get_latest_blockhash().await {
+                    *block_hash_rw.write().await = blockhash;
+                }
+
+                if let Ok(block_height) = rpc_client.get_block_height().await {
+                    current_block_height.store(block_height, std::sync::atomic::Ordering::Relaxed);
+                }
+                tokio::time::sleep(BLOCKHASH_REFRESH_RATE).await;
+            }
+        });
+    }
+
+    let transaction_map = Arc::new(DashMap::<Signature, TransactionData>::new());
+    let transaction_errors = Arc::new(Mutex::new(vec![None; messages.len()]));
+    let confirmed_transactions = Arc::new(AtomicU32::new(0));
+    // tasks which confirms the transactions that were sent
+    {
+        let exit_signal = exit_signal.clone();
+        let current_block_height = current_block_height.clone();
+        let transaction_map = transaction_map.clone();
+        let rpc_client = rpc_client.clone();
+        let confirmed_transactions = confirmed_transactions.clone();
+        let transaction_errors = transaction_errors.clone();
+        tokio::spawn(async move {
+            while !exit_signal.load(Ordering::Relaxed) {
+                if !transaction_map.is_empty() {
+                    // transaction is valid for MAX_PROCESSING_AGE
+                    // We only verify the transactions till MAX_RECENT_BLOCKHASHES which is 2*MAX_PROCESSING_AGE
+                    // If the transaction is not confirmed in MAX_RECENT_BLOCKHASHES we declare that it is expired.
+                    // give our CONFIRMATION_REFRESH_RATE of 5 secs we can have maximum 12.5 blocks
+                    // So there are 300 slots for the transactions to get confirmed
+                    let current_block_height = current_block_height.load(Ordering::Relaxed);
+                    let transactions_to_verify: Vec<Signature> = transaction_map
+                        .iter()
+                        .filter(|x| {
+                            current_block_height < x.blockheight + MAX_RECENT_BLOCKHASHES as u64
+                        })
+                        .map(|x| *x.key())
+                        .collect();
```

Comment:
I guess it would be fine to recheck status on transactions that have already expired.  It should definitely check status on for all the transactions that have not yet expired.